src/main/java/com/shapesecurity/shift/es2017/parser/token/EscapedKeywordToken.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2014 Shape Security, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.shapesecurity.shift.es2017.parser.token;
+
+import com.shapesecurity.shift.es2017.parser.SourceRange;
+import com.shapesecurity.shift.es2017.parser.TokenType;
+
+import javax.annotation.Nonnull;
+
+public class EscapedKeywordToken extends IdentifierToken {
```

Comment:
I don't think it's a good idea for this to subclass `IdentifierToken`. We treat these tokens as being effectively `enum`s, not classes with inheritance.